web/app.js

```diff
@@ -676,6 +681,13 @@ const PDFViewerApplication = {
     this.pdfViewer.currentScaleValue = DEFAULT_SCALE_VALUE;
   },
 
+  rescale(scaleFactor, options) {
+    if (this.pdfViewer.isInPresentationMode) {
+      return;
+    }
+    this.pdfViewer.rescale(scaleFactor, options);
+  },
+
```

Comment:
Do we *really* need this, i.e. can't we change the existing `zoomIn`/`zoomOut` methods to also support this use-case as well (and thus call those methods with pinch-zooming as well)?

```javascript
  zoomIn(steps, scaleFactor = null) {
    if (this.pdfViewer.isInPresentationMode) {
      return;
    }
    this.pdfViewer.increaseScale(steps, {
      drawingDelay: AppOptions.get("defaultZoomDelay"),
      scaleFactor,
    });
  },

  zoomOut(steps, scaleFactor = null) {
    if (this.pdfViewer.isInPresentationMode) {
      return;
    }
    this.pdfViewer.decreaseScale(steps, {
      drawingDelay: AppOptions.get("defaultZoomDelay"),
      scaleFactor,
    });
  },
```

And in the `pdf_viewer.js` file we'd make changes like the following:

```javascript
  increaseScale(steps = 1, options = null) {
    if (!this.pdfDocument) {
      return;
    }
    let newScale = this._currentScale;

    if (options?.scaleFactor > 1) {
      newScale = Math.min(
        MAX_SCALE,
        Math.ceil(newScale * options.scaleFactor * 100) / 100
      );
    } else {    
      do {
        newScale = (newScale * DEFAULT_SCALE_DELTA).toFixed(2);
        newScale = Math.ceil(newScale * 10) / 10;
        newScale = Math.min(MAX_SCALE, newScale);
      } while (--steps > 0 && newScale < MAX_SCALE);
    }
    options ||= Object.create(null);
    options.noScroll = false;
    this._setScale(newScale, options);
  }

  decreaseScale(steps = 1, options = null) {
    if (!this.pdfDocument) {
      return;
    }
    let newScale = this._currentScale;

    if (options?.scaleFactor > 0 && options?.scaleFactor < 1) {
      newScale = Math.max(
        MIN_SCALE,
        Math.floor(newScale * options.scaleFactor * 100) / 100
      );
    } else {
      do {
        newScale = (newScale / DEFAULT_SCALE_DELTA).toFixed(2);
        newScale = Math.floor(newScale * 10) / 10;
        newScale = Math.max(MIN_SCALE, newScale);
      } while (--steps > 0 && newScale > MIN_SCALE);
    }
    options ||= Object.create(null);
    options.noScroll = false;
    this._setScale(newScale, options);
  }
```